go/runtime/host/protocol/connection.go

```diff
@@ -171,6 +192,22 @@ func (c *connection) Call(ctx context.Context, body *Body) (*Body, error) {
 }
 
 func (c *connection) call(ctx context.Context, body *Body) (*Body, error) {
+	start := time.Now()
+	defer func() {
+		if viper.GetString(metrics.CfgMetricsMode) != metrics.MetricsModeNone {
+			var call string
+			// Determine request/response type by taking the first non-empty field.
```

Comment:
Could this part (extracting the name of the body request) be moved to the Body struct, e.g. a `String() string` method or something.